server/src/internalClusterTest/java/org/opensearch/gateway/remote/RemoteClusterStateCleanupManagerIT.java

```diff
@@ -108,6 +129,23 @@ public void testRemoteCleanupDeleteStale() throws Exception {
             .add("cluster-state")
             .add(getClusterState().metadata().clusterUUID());
         BlobPath manifestContainerPath = baseMetadataPath.add("manifest");
+
+        BlobStoreRepository routingTableRepository = (BlobStoreRepository) repositoriesService.repository(REPOSITORY_3_NAME);
+        List<IndexRoutingTable> indexRoutingTables = new ArrayList<>(getClusterState().routingTable().indicesRouting().values());
+        BlobPath indexRoutingBasePath = baseMetadataPath.add(INDEX_ROUTING_PATH_TOKEN);
+        RemoteStoreEnums.PathHashAlgorithm pathHashAlgo = RemoteStoreEnums.PathHashAlgorithm.FNV_1A_BASE64;
+        BlobPath indexRoutingPath = pathType.path(
+            RemoteStorePathStrategy.PathInput.builder()
+                .basePath(indexRoutingBasePath)
+                .indexUUID(indexRoutingTables.get(0).getIndex().getUUID())
+                .build(),
+            pathHashAlgo
+        );
+        assertBusy(() -> {
+            // There would be >=3 files as shards will transition from INIT -> UNASSIGNED -> STARTED state
+            assertTrue(routingTableRepository.blobStore().blobContainer(indexRoutingPath).listBlobs().size() >= 3);
+        });
```

Comment:
Why is this test in RemoteStateCleanupManagerIT rather than in RemoteRoutingTableServiceIT?

Reply:
As routing table is a part of cluster state added test in RemoteStateCleanupManagerIT.

Reply:
Routing table might not always be enabled. We need to ensure tests cover both scenarios.

Reply:
Addressed in revision.